### ISSUE TITLE
WIKI Updater tasks are failing with error in latest version 2.12.21

### ISSUE THREAD

**Issue by nanvik** [rfennell/AzurePipelines#1781]
### Azure DevOps Extensions

WIKI Updater

### Platform

Azure DevOps Services

### Extension Version

2.12.21

### Describe the bug

This task just started failing in all our pipelines. 2.11.32 was working fine. Been using the task without issue since January. Task was updated to 2.12.21 this morning, and all pipelines using this task are failing with the following error. Seems something is wrong with the minimatch node package used by this task.

Task is failing with error:

##[error]Unhandled: Unexpected token ;
##[error]C:\agent\_work\_tasks\WikiUpdaterTask_e59c5ae6-3afd-479d-bc40-81cd6c541840\2.12.21\node_modules\rimraf\node_modules\minimatch\dist\cjs\index.js:192

### Repo Steps

...
1. Have build pipeline generate a release notes markdown file using the Cross Platform Generate Release Notes task
2. Upload markdown file to Azure DevOps wiki using the WIKI Updater task.

### Expected Behavior

Should be no error and task should succeed.

### Logging Information

Starting: Git based WIKI File Updater
==============================================================================
Task         : Git based WIKI Single File Updater from Black Marble
Description  : A tools to update or create a file in a WIKI based in a Git repo 
Version      : 2.12.21
Author       : Richard Fennell
Help         : Version: 2.12.21. [More Information](https://github.com/rfennell/vNextBuild/wiki/Wiki-Updater-Task/)
==============================================================================
##[error]Unhandled: Unexpected token ;
##[error]C:\agent\_work\_tasks\WikiUpdaterTask_e59c5ae6-3afd-479d-bc40-81cd6c541840\2.12.21\node_modules\rimraf\node_modules\minimatch\dist\cjs\index.js:192
    options;
           ^

SyntaxError: Unexpected token ;
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (C:\agent\_work\_tasks\WikiUpdaterTask_e59c5ae6-3afd-479d-bc40-81cd6c541840\2.12.21\node_modules\rimraf\node_modules\glob\dist\commonjs\index.js:4:21)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
Finishing: Git based WIKI File Updater

**Comment by rfennell** [comment]
@nanvik your comment you are using Classic pipelines beat me to my initial response.

I will look this morning to see if I can spot the issue

**Comment by rfennell** [comment]
The recent change in this (and all my extensions) was to update all my dependent libraries, [this mostly addressed the testing frameworks](https://github.com/rfennell/AzurePipelines/commit/01d767ed5896b220681a6c53448cf7c2c2ba5581), with the exception in the case of this task to change to the newer sync version of RIMRAF which is used to delete the local repo folder prior to the git clone.

I did no see the [error in my CI/CD test](https://dev.azure.com/richardfennell/GitHub/_build/results?buildId=11689&view=logs&j=e7df25ac-5e39-50f2-81a8-2194340cf461&t=b5bedf99-2065-5685-9d63-f91974d8d2b6), so I am wondering if you a a combination of settings in us I don't test.

I have tried local test harness I have and have been able to run the task in a way that the `rimrafSync(localpath);` line is run without error.

So I can diagnose this further, can you please

1. Export the YAML for the task so I can see the exact settings in using
2. Run the pipeline with the variable `system.debug=true` set  or the the diagnostics checkbox ticked when the pipeline is run to get the detail debug log.

Sorry for the inconvenience 

Hopefully with that can repo the issue and fix it

**Comment by rfennell** [comment]
A bit more digging...

I suspect the issue is with your the format of the `localpath` parameter, as this it the only parameter passed to RIMRAF to tidy the folders. I have tested with both

- `$(System.DefaultWorkingDirectory)\repo`
- `c:\tmp\test\repo`

And have no problem. 

What are you using?

**Comment by z4kn4fein** [comment]
Hi @rfennell,

Today we started experiencing the same error. We are also using the classic pipeline, but this is our step's configuration in `yaml` format:

```yaml
steps:
- task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks.WikiUpdaterTask.WikiUpdaterTask@2
  displayName: 'Git based WIKI File Updater'
  inputs:
    repo: '<repo>'
    filename: ' release-notes/home-page/$(Release.ReleaseName).md'
    dataIsFile: true
    sourceFile: '$(System.DefaultWorkingDirectory)/releasenotes.md'
    message: 'Release notes created by pipeline'
    gitname: '$(Release.RequestedFor)'
    gitemail: '$(Release.RequestedForEmail)'
    useAgentToken: true
```

**Comment by rfennell** [comment]
@z4kn4fein  thanks for the logs, I now have a repo on my test system.

The issue is that on the failing systems the task is being run by the Node10 execution engine, but on the my systems (and the hosted runners) it is using Node16. If I force the execution to use Node10 I see the same error.

> Current [guidance](https://devblogs.microsoft.com/devops/node-runner-update-guidance-for-azure-pipelines-task-authors/) is to move everything over to Node16, and if Node16 is present it will be used over Node10 if both are present.

I am looking currently looking to see how to fix this in Node10.

However, can I ask why your what I assume are private agents don't have Node16 installed?

**Comment by z4kn4fein** [comment]
Yes, we are using a private agent, and updating the agent version to `3.x` solved the issue as it's now using node `v16`. Thank you!

**Comment by nanvik** [comment]
Ah! Yes, private agent here as well - I'll try updating the agent now and report back - I should have thought of trying that!

**Comment by nanvik** [comment]
Update agent and yes, all is good now. Thanks for the quick response Richard. These are great extensions!

**Comment by rfennell** [comment]
Great to hear you are sorted.

The route cause of the issues is I moved from rimraf 3.0.5 to 5.0.5 but this requires [Node 14 or later](https://github.com/isaacs/rimraf/issues/285). I did not find this in testing as all the own agents and the hosted ones I test against have Node16 installed.

Note, this will also effect the WIKIPDF exporter extension.

My plan is to revert this change (go back to older RIMRAF 3.025 package that supported Node 10) and release this as a 2.13.x  version.

I will then release a 3.x version that only supports Node16
